Assets/LeapMotion/Core/Scripts/Utils/Utils.cs

```diff
@@ -898,88 +824,96 @@ public struct ChildrenEnumerator : IEnumerator<Transform> {
       thisTransform.rotation = Quaternion.LookRotation(thisTransform.position - transform.position, upwards);
     }
 
+    #endregion
+
+    #region Vector3 Utils
+
     /// <summary>
-    /// Returns the rotation that makes a transform at fromPosition point its forward
-    /// vector at targetPosition and keep its rightward vector parallel with the horizon
-    /// defined by a normal of Vector3.up.
-    /// 
-    /// For example, this will point an interface panel at a user camera while
-    /// maintaining the alignment of text and other elements with the horizon line.
+    /// Rightward syntax for applying a Quaternion rotation to this vector; literally
+    /// returns byQuaternion * thisVector -- does NOT modify the input vector.
     /// </summary>
-    /// <returns></returns>
-    public static Quaternion FaceTargetWithoutTwist(Vector3 fromPosition,
-                                                    Vector3 targetPosition,
-                                                    bool flip180 = false) {
-      return FaceTargetWithoutTwist(fromPosition, targetPosition, Vector3.up, flip180);
+    public static Vector3 RotatedBy(this Vector3 thisVector, Quaternion byQuaternion) {
+      return byQuaternion * thisVector;
     }
 
     /// <summary>
-    /// Returns the rotation that makes a transform at fromPosition point its forward
-    /// vector at targetPosition and keep its rightward vector parallel with the horizon
-    /// defined by the upwardDirection normal.
-    /// 
-    /// For example, this will point an interface panel at a user camera while
-    /// maintaining the alignment of text and other elements with the horizon line.
+    /// Returns this position moved towards the argument position, up to but no more than
+    /// the max movement amount from the original position.
     /// </summary>
-    public static Quaternion FaceTargetWithoutTwist(Vector3 fromPosition,
-                                                    Vector3 targetPosition,
-                                                    Vector3 upwardDirection,
-                                                    bool flip180 = false) {
-      Vector3 objToTarget = targetPosition - fromPosition;
-      return Quaternion.LookRotation((flip180 ? -1 : 1) * objToTarget,
-                                     upwardDirection);
+    public static Vector3 MovedTowards(this Vector3 thisPosition,
+                                      Vector3 otherPosition,
+                                      float maxMovementAmount) {
+      var delta = thisPosition - otherPosition;
+      if (delta.sqrMagnitude > maxMovementAmount * maxMovementAmount) {
+        delta = Vector3.ClampMagnitude(delta, maxMovementAmount);
+      }
+      return thisPosition + delta;
     }
 
-    #endregion
-
-    #region Quaternion Utils
+    public static Vector3 ToVector3(this Vector4 v4) {
```

Comment:
There is already an implicit conversion from Vector4 to Vector3, so I don't think this method is necessary.

Reply:
In a situation where I'm going back and forth between Vector3s and Vector4s (say, for a normal-calculation algorithm on a mesh), I like the conversions to be explicit. Do you mind if we keep this? I'll add a comment that says that an implicit version exists so there's no confusion

Reply:
(This is particularly relevant if I'm doing a bunch of sequential calls like `MovedTowards(bluh).ToVector4().MyCustomVector4Extension().ToVector3().RotatedBy()...`)